### ISSUE TITLE
ym2151.c clipping

### ISSUE THREAD

**Issue by vampirefrog** [vgmrips/vgmplay#32]
Hello. When using the following voice with the ym2151 core:

```
{
/* AR D1R D2R  RR  D1L TL KS MUL DT1 DT2 AME */
   31,  0,  0, 15, 0,  0, 0,  1,  0,  0,  0,
   31,  0,  0, 15, 0,  0, 0,  1,  0,  0,  0,
   31,  0,  0, 15, 0,  0, 0,  1,  0,  0,  0,
   31,  0,  0, 15, 0,  0, 0,  1,  0,  0,  0,
/* CON FL OP */
   7,  0, 15
}
```

which is the loudest sine wave I could get, you get clipping. Try values 1,3,7 and 15 for OP (operator mask). This is due to the code:

```
	if (chanout[chan] > +16384)			chanout[chan] = +16384;
	else if (chanout[chan] < -16384)	chanout[chan] = -16384;
```
in [chips/ym2151.c:1755](https://github.com/vgmrips/vgmplay/blob/master/VGMPlay/chips/ym2151.c#L1755) and [chips/ym2151.c:1821](https://github.com/vgmrips/vgmplay/blob/master/VGMPlay/chips/ym2151.c#L1821)

If you compare to XM6 Pro-68k, this clipping does not occur. I don't think it happens on the real OPM either.

Changing FINAL_SH doesn't help either, it just makes the signal smaller but still clipped.

Please either remove the clipping check or define it as MINOUT and MAXOUT. Either way, define it as a macro instead of raw numbers.

Thanks!

**Comment by vampirefrog** [comment]
In fact in MAME it has been removed already https://github.com/mamedev/mame/blob/master/src/devices/sound/ym2151.cpp

**Comment by superctr** [comment]
We added that intentionally due to a song in X68K Space Harrier which had
supposedly intentional clipping.

**Comment by vampirefrog** [comment]
oh and don't forget to use MAXOUT and MINOUT either in the chip or in the mixing code. perhaps use soft clipping.
For particular games that need clipping, perhaps add a VGM param that would emulate the opamp saturation.

**Comment by ValleyBell** [comment]
Someone on IRC confirmed that the YM2151 doesn't clip a while ago.  
Per-channel clipping was an addition made by me based on the wrong assumption that the YM2151 clips like the YM2612 does.

I forgot to fix the bug for v0.40.9, but it is fixed in libvgm (by not clipping at all anywhere), so I'll consider this issue solved.
